src/containers/GroceryFormWhere.js

```diff
@@ -0,0 +1,218 @@
+
+const validate = (val) => {
+  if (!isValidPhoneNumber(val)) {
+    return 'Please enter a valid phone number';
+  }
+};
+
+const validateEmail = (val) => {
+  if (!isValidEmail(val)) {
+    return 'Please enter a valid email address';
+  }
+};
+
+// service todo: Temp values while we build backend or an enum for this part.
+const LANGUAGES = [
+  { label: 'English', value: 'english' },
+  { label: 'Spanish', value: 'spanish' },
+];
+
+function GroceryFormWhere() {
+  const history = useHistory();
+  const { t } = useTranslation();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [addAdditionalContact, setAddAdditionalContact] = useState(false);
+  const [fields, setFields] = useState({
+    zipCode: '',
+    neighborhood: '',
+    crossStreet: '',
+  });
+
+  const handleFieldChange = useCallback(
+    (field) => (value) => {
+      setFields((fields) => ({
+        ...fields,
+        [field]: value,
+      }));
+    },
+    [],
+  );
+
+  const handleSubmit = () => {
+    setIsLoading(true);
+    // service todo: wire-up form events
+    history.push(routeWithParams(Routes.SERVICE_GROCERIES_WHEN));
+  };
+
+  const fieldData = [
+    {
+      customOnChange: handleFieldChange('zipCode'),
+      label: t('service.grocery.where.labels.zip'),
+      name: 'zipCode',
+      type: formFieldTypes.INPUT_TEXT,
+      value: fields.zipCode,
+    },
+    {
+      customOnChange: handleFieldChange('neighborhood'),
+      label: t('service.grocery.where.labels.neighborhood'),
+      // service todo: wire-up neighborhoods data
+      options: neighborhoods.Brooklyn,
+      name: 'neighborhood',
+      type: formFieldTypes.INPUT_DROPDOWN,
+      value: fields.neighborhood,
+    },
+    {
+      customOnChange: handleFieldChange('crossStreet'),
+      label: t('service.grocery.where.labels.crossStreet'),
+      name: 'crossStreet',
+      type: formFieldTypes.INPUT_TEXT,
+      value: fields.crossStreet,
+    },
+    {
+      type: formFieldTypes.NODE,
+      node: [
+        <Fragment key="additional">
+          {addAdditionalContact && (
+            <div
+              css={{
+                borderTop: Borders.GRAY1,
+                padding: `${Space.S30}px 0`,
+                width: '100%',
+              }}
+            >
+              <Text type={TEXT_TYPE.HEADER_4}>
+                {t('service.grocery.where.additionalContact.title')}
+              </Text>
+            </div>
+          )}
+
+          {!addAdditionalContact && (
+            <Text
+              onClick={(e) => {
+                e.preventDefault();
+                setAddAdditionalContact(true);
+              }}
+              as="button"
+              type={TEXT_TYPE.BODY_2}
+              css={[
+                textStyles.BOLD,
+                { color: Color.PRIMARY, display: 'flex', textAlign: 'left' },
+              ]}
+            >
+              <Plus
+                css={{
+                  flexShrink: 0,
```

Comment:
can we move these styles to a styles file?

Reply:
👍

Reply:
Moved this into a component, `AddiontalContact`, with styles, etc.